Index: hal/osal/rt/osal.h
===================================================================
--- hal/osal/rt/osal.h	(revision 6843)
+++ hal/osal/rt/osal.h	(working copy)
@@ -132,9 +132,9 @@
 /*===========================================================================*/
 
 /* Temporary types provided for ChibiOS 2.x compatibility.*/
-typedef io_queue_t GenericQueue;
-typedef input_queue_t InputQueue;
-typedef output_queue_t OutputQueue;
+//typedef io_queue_t GenericQueue;
+//typedef input_queue_t InputQueue;
+//typedef output_queue_t OutputQueue;
 typedef bool bool_t;
 #define OSAL_FREQUENCY OSAL_ST_FREQUENCY
 
Index: rt/ports/ARMCMx/compilers/GCC/chcoreasm_v6m.s
===================================================================
--- rt/ports/ARMCMx/compilers/GCC/chcoreasm_v6m.s	(revision 6843)
+++ rt/ports/ARMCMx/compilers/GCC/chcoreasm_v6m.s	(working copy)
@@ -85,12 +85,12 @@
                 .thumb_func
                 .globl  _port_thread_start
 _port_thread_start:
-#if CH_DBG_SYSTEM_STATE_CHECK
+/*#if CH_DBG_SYSTEM_STATE_CHECK
                 bl      _dbg_check_unlock
 #endif
 #if CH_DBG_STATISTICS
                 bl      _stats_stop_measure_crit_thd
-#endif
+#endif*/
                 cpsie   i
                 mov     r0, r5
                 blx     r4
@@ -104,36 +104,36 @@
                 .thumb_func
                 .globl  _port_switch_from_isr
 _port_switch_from_isr:
-#if CH_DBG_STATISTICS
+/*#if CH_DBG_STATISTICS
                 bl      _stats_start_measure_crit_thd
 #endif
 #if CH_DBG_SYSTEM_STATE_CHECK
                 bl      _dbg_check_lock
-#endif
+#endif */
                 bl      chSchDoReschedule
-#if CH_DBG_SYSTEM_STATE_CHECK
+/*#if CH_DBG_SYSTEM_STATE_CHECK
                 bl      _dbg_check_unlock
 #endif
 #if CH_DBG_STATISTICS
                 bl      _stats_stop_measure_crit_thd
-#endif
+#endif*/
                 .globl  _port_exit_from_isr
 _port_exit_from_isr:
                 ldr     r2, .L2
                 ldr     r3, .L3
                 str     r3, [r2, #0]
-#if CORTEX_ALTERNATE_SWITCH
+/*#if CORTEX_ALTERNATE_SWITCH
                 cpsie   i
-#endif
+#endif*/
 .L1:            b       .L1
 
                 .align  2
 .L2:            .word   SCB_ICSR
-#if CORTEX_ALTERNATE_SWITCH
+/*#if CORTEX_ALTERNATE_SWITCH
 .L3:            .word   ICSR_PENDSVSET
-#else
+#else*/
 .L3:            .word   ICSR_NMIPENDSET
-#endif
+/*#endif*/
 
 #endif /* !defined(__DOXYGEN__) */
 
